tool/plugin/lib/plugin.dart

```diff
@@ -657,20 +753,57 @@ class DeployCommand extends ProductCommand {
 /// Generate the plugin.xml from the plugin.xml.template file.
 /// If the --release argument is given, create a git branch and
 /// commit the new file to it, assuming the release checks pass.
-class GenCommand extends Command {
+/// Note: The product-matrix.json file includes a build spec for the
+/// EAP version at the end. When the EAP version is released that needs
+/// to be updated.
+class GenCommand extends ProductCommand {
   final BuildCommandRunner runner;
 
   GenCommand(this.runner);
 
   String get description =>
-      'Generate a valid plugin.xml and .travis.yaml for the Flutter plugin.\n'
+      'Generate a valid plugin.xml and .travis.yml for the Flutter plugin.\n'
       'The plugin.xml.template and product-matrix.json are used as input.';
 
   String get name => 'gen';
 
-  Future<int> run() async {
-    // TODO(messick): Implement GenCommand.
-    throw 'unimplemented';
+  Future<int> doit() async {
+    List json = readProductMatrix();
+    var spec = new SyntheticBuildSpec.fromJson(json.first, release, json.last);
```

Comment:
Similarly here.

Reply:
I'll clean these up later. I'll make a pass through all the code looking for this inconsistency, now that you've brought it to my attention.